### PR TITLE
[RFC] Remove the 'compatible' option from runtime files

### PR DESCRIPTION
**Pull request by Grimy**
This removes most uses of the `'compatible'` options from runtime files : 
- Whenever it is read in a VimL script, the code is simplified by assuming it is false (this sometimes lead to removing entire branches).
- Many mentions of this option were removed from the documentation (there were a lot, usually of the form “this won’t work if compatible is set, so just set nocp”).
- Vi-compatibility was removed from the list of design goals in runtime/doc/develop.txt (this file should probably be removed altogether, since Neovim has a separate “Contributing” page on the wiki).
- The documentation for the -N and -C command-line options that are used to control `'compatible'` is also removed. (Maybe document -N as ignored for compatibility? OTOH, -C should definitely not be documented, since it’s now an error).
- The file `options.txt` is left untouched by this PR, so as to not conflict with #1626.

Even though those are mostly trivial changes, this makes for a huge PR, so let me know if you think I should split it.

I signed the CLA.

### CODE REVIEW

**Comment by justinmk** [comment]
> Even though those are mostly trivial changes, this makes for a huge PR, so let me know if you think I should split it.

No, let's keep it small. thanks.

**Comment by Pyrohh** [comment]
I wouldn't bother with the `netrw` related items, which for this PR would be:

- `runtime/autoload/netrw*.vim`
- `runtime/plugin/netrwPlugin.vim`
- `runtime/doc/pi_netrw.txt`

You'd have to ask @justinmk, but IIRC he said that he doesn't want to maintain a fork of it since it will eventually be dropped (hopefully).

File: runtime/doc/intro.txt [hunk 1]
```diff
@@ -500,8 +500,7 @@ the ":map" command.  The rules are:
 	   <t_kd>		"kd" termcap entry (cursor down key)
 
 If you want to use the full <> notation in Vim, you have to make sure the '<'
-flag is excluded from 'cpoptions' (when 'compatible' is not set, it already is
-by default). >
+flag is excluded from 'cpoptions'.
```
Comment (Pyrohh) [inline runtime/doc/intro.txt]: Looks like the `>` was removed. You can use this to toggle conceal levels in order see hidden chars like that:
```
nnoremap coc :set conceallevel=<C-R>=&conceallevel == 2 ? 0 : 2<CR><CR>
```

File: runtime/doc/pi_netrw.txt [hunk 2]
```diff
@@ -118,10 +118,9 @@ local browsing easy!  First, make sure that you have plugins enabled, so
 you'll need to have at least the following in your <.vimrc>:
 (or see |netrw-activate|) >
 
-	set nocp                    " 'compatible' is not set
 	filetype plugin on          " plugins are enabled
 <
-(see |'cp'| and |:filetype-plugin-on|)
+(see ||:filetype-plugin-on|)
```
Comment (Pyrohh) [inline runtime/doc/pi_netrw.txt]: Extra `|` after `see`.

**Comment by Pyrohh** [comment]
Also regarding #1626, I'm sure you could ping @klusark and pull in their changes if they don't respond (assuming you want to).

**Comment by Grimy** [comment]
Thanks for your comments. I added the '>', removed the '|' and rebased on the new master. Should I revert my changes on netrw?

**Comment by Pyrohh** [comment]
Sure, that's probably a good idea.

**Comment by Grimy** [comment]
I amended out all changes to the netrw files you mentionned.

**Comment by Pyrohh** [comment]
@Grimy you can probably squash this into two commits:
```
Remove '(no)compatible' references: docs
Remove '(no)compatible' references: runtime
```

**Comment by Pyrohh** [comment]
I'm not sure if you mean to do this on purpose, but there are still many references to `Vi compatible` (and other mentions of `compatible` and `nocompatible`).

**Comment by Pyrohh** [comment]
@Grimy do you plan on finishing this, or may I finish it for you? I won't change the author date.

**Comment by Grimy** [comment]
Feel free to finish this. Thanks for asking (:

**Comment by Pyrohh** [comment]
Thanks, btw when I said I won't change the author date, I meant the author name. Now, if you or @justinmk wouldn't mind closing this, that'd be great.

**Pull request closed without merging**